testutil/compose: add readme (#593)
ObolNetwork/charon: testutil/compose/README.md

# Charon Compose

> Run, test, and debug a developer-focussed insecure local charon cluster using docker-compose

Compose is a tool that generates `docker-compose.yml` files such that different charon clusters can be created and run.

The aim is for developers to be able to debug features and check functionality of clusters on their local machines.

> The `compose` command should be executed in a sequential steps:

Reviewer: ```suggestion
The `compose` command should be executed in sequential steps:
```

 1. `compose clean`: Cleans the compose directory of existing artifacts.

>  1. `compose define`: Define the target cluster and how keys are to be created.

Reviewer: ```suggestion
 2. `compose define`: Defines the target cluster and how keys are to be created.
```
The numbering is incorrect

    1. It outputs `config.json` which is the compose config
    1. It also creates `docker-compose.yml` in order to create `cluster-definition.json` if `keygen==dkg`.
 1. `compose lock`: Creates `docker-compose.yml` to create threshold key shares and the `cluster-lock.json` file.
 1. `compose run`: Creates `docker-compose.yml` to run the cluster.

Note that compose automatically runs `docker-compose up` at the end of each command. This can be disabled via `--up=false`.

The `compose define` step configures the target cluster and key generation process, it supports the following flags:
 - `--keygen`: Key generation process: `create` or `dkg`.
   - Either `create` to create keys locally via `charon create cluster`
   - or `dkg` to create keys via `charon create dkg` followed by `charon dkg`.
 - `--split-keys-dir`: Path to a folder containing keys to split. Only applicable to `--keygen=create`.

>  - `--build-local`: Build a local charon binary from source. Note this requires the `CHARON_REPO` path env var. Devs are encourages to put this in the bash profile.

Reviewer: ```suggestion
 - `--build-local`: Build a local charon binary from source. Note this requires the `CHARON_REPO` path env var. Devs are encouraged to put this in the bash profile.
```
